doc/modules/svm.rst

```diff
@@ -689,6 +692,174 @@ term :math:`\rho`
    Volume 14 Issue 3, August 2004, p. 199-222. 
 
 
+.. _svm_one_class_svm:
+
+One-Class SVM
+-------------
+
+This model, proposed by Schölkopf et al. (2001), estimates the support
+of a high-dimensional distribution by constructing a supporting hyperplane
+in the feature space corresponding to the kernel, which effectively
+separates the data set from the origin with maximum margin.
+
+For the training sample :math:`(x_i)_{i=1}^l` with weights :math:`(w_i)_{i=1}^l`,
+:math:`\sum_{i=1}^l w_i>0`, the One-Class SVM solves the following primal problem:
+
+
+.. math::
+
+   \min_{\rho,\xi,w} \frac12 w^Tw - \rho + \frac{1}{\nu W} \sum_{i=1}^l w_i \xi_i \,, \\
+
+   \textrm {subject to } & w^T\phi(x_i) \geq \rho - \xi_i \,, \\
+                         & \xi_i \geq 0\,,\, i=1, \ldots, l \,,
+
+
+where :math:`\phi(\cdot)` is the feature map associated with the
+kernel :math:`K(\cdot,\cdot)`, and :math:`W = \sum_{i=1}^l w_i`.
+
+The dual problem is
+
+
+.. math::
+
+   \min_\alpha \frac12 \alpha^T Q\alpha\,\\
+
+   \textrm {subject to } & 0\leq \alpha_i \leq w_i\,,\, i=1, \ldots, l \,,\\
+                         & e^T\alpha = \nu W \,,
+
+
+where :math:`e\in \mathbb{R}^{l\times 1}` is the vector of ones and
+:math:`Q_{ij} = K(x_i, x_j)` is the kernel Gram matrix.
+
+The optimal decision function is given by:
+
+.. math:: x\mapsto \operatorname{sgn}(\sum_{i=1}^l \alpha_i K(x_i, x) - \rho) \,,
+
+where :math:`+1` indicates and inliner and :math:`-1` -- an outlier.
+
+The parameter :math:`\nu\in(0,1]` determines the fraction of outliers
+in the training dataset. More technically :math:`\nu` is:
+  * an upper bound on the fraction of the training points lying outside
+    the estimated region;
+
+  * a lower bound on the fraction of support vectors.
+
+.. topic:: References:
+
+  * `Estimating the support of a high-dimensional distribution
+    <http://dl.acm.org/citation.cfm?id=1119749>`_ Schölkopf,
+    Bernhard, et al. Neural computation 13.7 (2001): 1443-1471.
+    doi:10.1162/089976601750264965
+
+
+.. _svm_svdd:
+
+SVDD
+----
+
+Support Vector Data Description (SVDD), proposed by Tax and Duin (2004),
+aims at finding spherically shaped boundary around a data set. The original
+formulation suffered from non-convexity issues related to optimality of
+the attained solution for certain values of regularization parameter :math:`C`.
+Chang, Lee, and Lin (2013) suggested a reformulation of the SVDD model
+which had a well-defined and provably unique global solution for any :math:`C>0`.
+
+Scikit's SVDD model is a modified version of the 2013 SVDD formulation.
```

Comment:
Yes, the implementation I propose includes sample weights and uses `nu` instead of `C`.

It was pretty straightforward to add weights to the 2013 SVDD: I just had to check if the key theorems hold, and what are their requirements, and work on the constraints in the dual problem a little bit.